tools/devops/automation/templates/common/mac-agent-logs.yml

```
@@ -0,0 +1,71 @@
+# This template will gather all the logs of a mac agent from a given starting poing in time
+# to a given end point. The temaplte also takes the path for the file where the logs will be written.
+#
+
+parameters:
+
+- name: workingDirectory
+  type: string
+  default: ''
+
+- name: outputPath 
+  type: string 
+  default: ''
+
+- name: predicate
+  type: string
+  default: ''
+
+- name: style
+  type: string
+  default: 'compact'
+  values:
+    - default
+    - compact
+    - json
+    - syslog
+
+steps:
+
+- pwsh: |
+    $style=$Env:STYLE
+    $output=$Env:OUTPUT
+    if (-not $output) {
+      Write-Host "Not output path provided using default path: agent-logs.log"
+      $output="agent-logs.log"
+    } else {
+      Write-Host "output path: $output"
+    }
+
+    # there is not var that gives us the start time, but we can use the rest api to get the time when
```

```suggestion
    # there is no var that gives us the start time, but we can use the rest api to get the time when
```
or maybe "not a var"